stock-filters/ChirpVillage/Builder.py

```diff
@@ -52,9 +53,11 @@ def construct_pointed_roof(level, coords, biome, height):
 def construct_pillars(level, coords, biome, height):
     minx, minz, miny, maxx, maxz = get_coords(coords)
     pillar_block = BlockUtils.get_beam_block(biome)
-    for y in range(0, height):
+    for y in range(height, 0, -1):
         for x in [minx, maxx-1]:
             for z in [minz, maxz-1]:
+                if y < miny and level.blockAt(x, y-2, z) in surface_blocks:
```

Comment:
If it is in surface blocks, you should replace it with a piller. Surface blocks contains air, grass, trees, etc

Reply:
so I think this should read not in surface_blocks

Reply:
My bad

Reply:
Additionally since water blocks are not included in surface_blocks, and you want to have water stilts you will need to not in do surface_blocks or equal to 9 (still water)